File: python/gemini/p130.py
from flask import Flask, request, render_template_string

app = Flask(__name__)

def render_response_from_env(env):
    # 1. Get name from request
    name = request.args.get('name', 'Guest')
    # 2. Get template from environment
    template = env.from_string("Hello, {{ name }}!")
    # 3. Render and return
    return template.render(name=name)

@app.route('/')
def home():
    # Pass the Jinja environment (app.jinja_env)
    return render_response_from_env(app.jinja_env)

File: c/gemini/p130.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void render_response_from_env(char* query_string) {
    char name[50] = "Guest";
    
    // 1. Extract name from query string (e.g., "name=Bob")
    if (query_string != NULL && strstr(query_string, "name=")) {
        sscanf(strstr(query_string, "name="), "name=%s", name);
    }

    // 2. The "Template" (simulated via printf in this env)
    printf("Content-Type: text/html\n\n");
    printf("<h1>Hello, %s!</h1>", name); // 3. Render
}

int main() {
    // Call with the QUERY_STRING environment variable
    render_response_from_env(getenv("QUERY_STRING"));
    return 0;
}